change-streams/main.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func ChangeStream(waitGroup sync.WaitGroup, stream *mongo.ChangeStream) {
+	defer stream.Close(context.TODO())
+	defer waitGroup.Done()
+	for stream.Next(context.TODO()) {
```

Comment:
If `stream.Next` returns false, you should check `stream.Err()` and panic as you do in other error cases. A change stream should not return false in a non-error case because it's a tailable cursor that stays alive indefinitely.

Reply:
Are you suggesting I should do something like the following?:

```
func iterateChangeStream(waitGroup sync.WaitGroup, stream *mongo.ChangeStream) {
	defer stream.Close(context.TODO())
	defer waitGroup.Done()
	for stream.Next(context.TODO()) {
		var data bson.M
		if err := stream.Decode(&data); err != nil {
			panic(err)
		}
		fmt.Printf("%v\n", data)
	}
	if stream.Err() != nil {
		panic(stream.Err())
	}
}
```

Reply:
Yeah, something like this should work.